subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyReportTaskIntegrationTest.groovy

```diff
@@ -1053,13 +1047,34 @@ compileClasspath - Compile classpath for source set 'main'.
             }
         """
 
-        when:
-        run ":a:dependencies"
+        executer.expectDocumentedDeprecationWarning("The compile configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 9.0. Please use the implementation configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations")
 
-        then:
-        !output.contains("\ncompile\n")
+        expect:
+        succeeds ':a:dependencies'
+    }
+
+    def "adding declarations to invalid configurations for declaration will fail"() {
+        given:
+        file("settings.gradle") << "include 'a', 'b'"
+
+        buildFile << """
+            subprojects {
+                configurations {
+                    compile.canBeDeclaredAgainst = false
+                    'default' { extendsFrom compile }
+                }
+                group = "group"
+                version = 1.0
+            }
+            project(":a") {
+                dependencies { compile project(":b") }
+            }
+        """
+        expect:
+        fails ':a:dependencies'
```

Comment:
Should we test for the error message here?

Reply:
Good idea.